### DISCUSSION TITLE
Bundle add-on

### DISCUSSION THREAD

**Discussion by swurzinger** [pretix/pretix#3672] [Q&A] [answered]
I am trying to bundle a free promotional add-on for a product, however this doesn't seem to work with the current version of pretix. Whenever I try to add such a product to the cart it complains about some products being temporarily unavailable.

Intended use case:
As a promotion, I want to bundle a free drink with every sold ticket for a limited time. The user shouldn't need to tick the addon himself.

Adding a regular product worked, but this has several downsides:
 - single cart entries (e.g. instead of 4 tickets with 4 free drinks we get 4 cart entries with  1 ticket + 1 drink each)
 - not shown in the check-in app, would need an extra physical ticket / QR-code

I'm also looking forward to having a functionality as described by another user in https://github.com/pretix/pretix/discussions/3526 

Should this already work and am I doing something wrong? (documentation didn't help, unfortunately, although it is great!)
If not: 
Is there a solution with existing features?
Could you suggest an approach to implement this and maybe give a few hints how to get into the code?

Thank you!

**Comment by raphaelm** [comment]
I think you are mixing bundled products and add-on products too hard? If you create it as a bundle product it does not need to be an addon product at the same time

**Reply by swurzinger** [reply]
Thank You for the response. I'm not sure if I understand you suggestion, though. Do you suggest to either (1) make it an addon (and not bundling it) or (2) make it a regular product and bundle it?

If so, both has undesirable affects, if I understood the pretix documentation correctly:
(1)
- an addon can't be bundled, the user actively has to select it
- the user can also deselect it and order without the addon (addon should be required)
(2)
- additional regular products need extra physical tickets/QR codes and are not shown / visible in the check-in app otherwise
- cart entries with quantities > 1 are split into single cart entries with quantity == 1 when regular products are bundled. The user can't simply change quantity, but has to delete each single product from the cart (or clear the cart and start from beginning)

Please correct me if any of that is wrong. I'd also be happy to hear about possible workarounds, if any. Thanks!

**Reply by raphaelm** [reply, marked as answer]
> an addon can't be bundled, the user actively has to select it

correct

> the user can also deselect it and order without the addon (addon should be required)

incorrect, you can require a minimum number of add-ons

> additional regular products need extra physical tickets/QR codes and are not shown / visible in the check-in app otherwise

a bundled product is treated like an addon product for all check-in purposes

> cart entries with quantities > 1 are split into single cart entries with quantity == 1 when regular products are bundled

correct

> The user can't simply change quantity, but has to delete each single product from the cart (or clear the cart and start from beginning)

hm, i think if I understand you correctly, this is incorrect and the user can do this

**Reply by swurzinger** [reply]
Thanks a lot for the detailed answers - I think the pros/cons of each the options are clear now.
So the behaviour I initially intended is indeed not fully supported, maybe take this as an idea for a new feature: allow making addons preselected/required.

In the end I'll probably just create dedicated products "ticket" and "ticket with free drink" and avoid using addons/bundles altogether because they both do not really fit my needs, as I've just learned 😅.
